src/hooks/usePrevious.ts

```diff
@@ -0,0 +1,13 @@
+import { useEffect, useRef } from "react";
+
+export function usePrevious<T>(value: T): [T | undefined, (value: T) => void] {
+  const ref = useRef<T>();
+
+  useEffect(() => {
+    ref.current = value;
+  });
+
+  const setValue = (value: T) => (ref.current = value);
```

Comment:
I don't think there should be a setter for this hook, it makes sense to be read-only value.

https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state